File: 203-remove-linked-list-elements/remove-linked-list-elements.java
class Solution {
    public ListNode removeElements(ListNode head, int val) {
       ListNode c= new ListNode(-1);
       ListNode tempC = c;
       ListNode temp = head;
       while(temp!=null){
        if(temp.val!=val){
            tempC.next = temp;
            tempC =temp;
        }
        temp = temp.next;
       } 
       tempC.next = null;
       return c.next;
    }
}

File: 203-remove-linked-list-elements/remove-linked-list-elements.c
#include <stdlib.h>

struct ListNode* removeElements(struct ListNode* head, int val) {
    if (head == NULL) {
        return NULL;
    }
    head->next = removeElements(head->next, val);
    if (head->val == val) {
        struct ListNode* next = head->next;
        free(head);
        return next;
    } else {
        return head;
    }
}
